packages/react-ui/src/components/ui/data-table.tsx

```diff
@@ -66,6 +66,27 @@ export function DataTable<TData, TValue>({
   const [deletedRows = [], setDeletedRows] = useState<TData[]>([]);
   const [loading, setLoading] = useState<boolean>(true);
 
+  const fetchDataAndUpdateState = async (params: URLSearchParams) => {
+    setLoading(true);
+    setTableData([]);
+    try {
+      const response = await fetchData(params);
+      const newData = response.data.map((row) => ({
+        ...row,
+        delete: () => {
+          setDeletedRows([...deletedRows, row]);
+        },
+      }));
+      setTableData(newData);
+      setNextPageCursor(response.next ?? undefined);
+      setPreviousPageCursor(response.previous ?? undefined);
+    } catch (error) {
+      console.error('Error fetching data:', error);
```

Comment:
Make it Set error on the table or show something, the user doesn't see the console log